analyzers/src/SonarAnalyzer.Common/Rules/Utilities/SymbolReferenceAnalyzerBase.cs

```diff
@@ -82,7 +82,7 @@ protected sealed override SymbolReferenceInfo CreateMessage(UtilityAnalyzerParam
                             .GetOrAdd(currentDeclaration.Symbol, _ => new List<ReferenceInfo>())
                             .Add(currentDeclaration);
                         knownNodes.Add(currentDeclaration.Node);
-                        knownIdentifiers.Add(currentDeclaration.Identifier.ValueText);
+                        knownIdentifiers.Add(currentDeclaration.Identifier.Text);
```

Comment:
This (probably) doesn't work for escape sequences. It may also not work for escaped non-keywords

[sharplab.io](https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQGED2AbbEDGMAlpgHYAyRMECU2yANDCFMjAD4ACATAIwCwAKE4BmAAQ8x6MQG8hYhRPGcALGICyACmCYcYgDpwADEYBspgJRiA9NbEALGDAAOyELYDuXgHSt8RIlR8TAATCG9ggFtrfHsoWkIaazB5RTlBRUyxMG8AFUwAZRgEIlIAc00LAG5UhQBfWrFG0Qk1dV5tXWwJTlJMAGsIAE8PTAQQi0b0rMU+wZGxkLzC4tKK6sa6zKE6oA)



Maybe it is better to use ValueText everywhere.

Reply:
Good catch! Indeed, `Text` did not worked correctly with escape sequences.
I have added tests for them, as well as for escaped non-keywords, and switched to `ValueText`.